Timber added

jafjdev opened the pull request:
Implementation of Timber working.
We're not using Log anymore.

Fix #55

Changes:

We're using Timber instead of Log.
https://github.com/JakeWharton/timber

jafjdev committed:
Timber added

Timber added and we're using timber instead of Log

dr0pdb commented on app/src/main/java/com/iitp/njack/iitp_connect/data/database/DatabaseContentProvider.java:
```diff
@@ -93,6 +94,7 @@ public Uri insert(@NonNull Uri uri, @Nullable ContentValues values) {
         return returnUri;
     }
 
+    @SuppressLint("TimberArgCount")
```
Don't suppress it.

dr0pdb commented on app/src/main/java/com/iitp/njack/iitp_connect/data/database/DatabaseContentProvider.java:
```diff
@@ -103,7 +105,7 @@ public int delete(@NonNull Uri uri, @Nullable String selection, @Nullable String
             case CONTESTS:
                 itemsDeleted = db.delete(DatabaseContract.ContestEntry.TABLE_NAME_CONTESTS, null, null);
-                Log.v("ContentProvider", "All the previous contests deleted i.e. " + itemsDeleted + " contests");
+                Timber.v("ContentProvider", "All the previous contests deleted i.e. " + itemsDeleted + " contests");
```
In order to avoid the ```TimberArgCount```, create a String above it with the concatenated values and use that.

jafjdev replied:
Oh, sorry , omw!

jafjdev replied:
i already tried :

String concatMsg = "All the previous contests deleted i.e. " + itemsDeleted + " contests";
Timber.v("ContentProvider",concatMsg);

but i'm still getting the suppress suggestion

dr0pdb replied:
```Timber.v(concatMsg);```

dr0pdb replied:
The TimberArgCount suggestion was because you're passing two arguments in it.

dr0pdb commented:
Moreover can you create an application class and initialize Timber in that class instead of every Activity ?

jafjdev commented:
i only did         
Timber.plant(new Timber.DebugTree()); 
On the HomeActivity.
Should i create a new class to Initialize timber?

dr0pdb commented:
> Should i create a new class to Initialize timber?

Not a new class. Override the application class. In case you don't know what an application class is: 
https://github.com/codepath/android_guides/wiki/Understanding-the-Android-Application-Class

jafjdev committed:
Fixed

Fixed supression , using Timber.

dr0pdb commented on app/src/main/java/com/iitp/njack/iitp_connect/core/androidapp/AndroidApplication.java:
```diff
@@ -0,0 +1,15 @@
+
+import timber.log.Timber;
+
+public class AndroidApplication extends Application {
```
Rename it to ```IITPConnectApplication```

dr0pdb commented on app/src/main/java/com/iitp/njack/iitp_connect/core/androidapp/AndroidApplication.java:
```diff
@@ -0,0 +1,15 @@
+    public void onCreate(){
+        super.onCreate();
+        Timber.plant(new Timber.DebugTree());
+
```
Remove this extra line

dr0pdb commented on app/src/main/java/com/iitp/njack/iitp_connect/core/home/HomeActivity.java:
```diff
@@ -17,6 +18,8 @@
 import com.iitp.njack.iitp_connect.R;
 import com.iitp.njack.iitp_connect.core.codingcalendar.views.CodingCalendarHomeActivity;
 
+import timber.log.Timber;
```
Remove this now. Optimize the imports using ```ctrl+alt+o```

dr0pdb commented on app/src/main/java/com/iitp/njack/iitp_connect/core/codingcalendar/adapters/CodingCalendarAdapter.java:
```diff
@@ -87,13 +90,14 @@ public void bind(int position) {
                 //get the platform of the contest.
                 String platformString = urlPlatform.getHost();
-                Log.v("ContestRecyclerViewAdap", platformString);
+                Timber.v("ContestRecyclerViewAdap", platformString);
```
Do the same thing here also which you did in ```DatabaseContentProvider``` i.e. use ```Timber.v(platformString);```.

dr0pdb commented on app/src/main/java/com/iitp/njack/iitp_connect/core/codingcalendar/adapters/CodingCalendarAdapter.java:
```diff
@@ -87,13 +90,14 @@ public void bind(int position) {
             contestStartTime.setText(contestStartTimeString);
         }
 
+        @SuppressLint("TimberArgCount")
```
Remove it.

dr0pdb commented on app/src/main/java/com/iitp/njack/iitp_connect/core/codingcalendar/views/CodingCalendarHomeActivity.java:
```diff
@@ -128,7 +129,7 @@ public Cursor loadInBackground() {
                     return getContentResolver().query(DatabaseContract.ContestEntry.CONTENT_URI_CONTESTS, null, null, null, null);
                 } catch (Exception e) {
-                    Log.e(TAG, "Failed to asynchronously load data.");
+                    Timber.e(TAG, "Failed to asynchronously load data.");
```
Remove the TAG and only pass in one argument. Also delete the TAG variable from this file

jafjdev committed:
Fixed

Removed extra lines, class renamed to IITPConnectApplication, imports optimized,SuppressLint removed.

dr0pdb reviewed (approved):
Nice work !
